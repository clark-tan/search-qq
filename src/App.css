.App {
  padding: 25px;
  margin: 50px auto 0;
  width: 340px;
  border-radius: 20px;
  box-shadow: 0 4px 30px 0 rgb(4 59 85 / 10%);
}

.title {
  font-size: 20px;
  margin-bottom: 10px;
  font-weight: bold;
}

.input {
  width: 100%;
  line-height: 24px;
  padding: 0 5px;
  margin-bottom: 20px;
  box-sizing: border-box;
  background-color: #fff;
  background-image: none;
  border: none;
  border-bottom: 1px solid #d9d9d9;
  border-radius: 2px;
  transition: all .3s;
}

.input:focus{
  border-color: #1890ff;
  border-right-width: 1px;
  outline: 0;
}

.result {
  display: flex;
  align-items: center;
  padding: 10px;
  border: 1px solid #ddd;
  overflow: hidden;
  height: 60px;
}

.nodata{
  background: url(../src/nodata.svg);
  background-size: contain;
  width: 60px;
  height: 60px;
  margin: auto;
}

.avatar {
  width: 60px;
  height: 60px;
}
.message {
  flex: 1 1;
  display: flex;
  padding: 10px;
  flex-direction: column;
  justify-content: center;
}
.name {
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  flex-grow: 0;
  width: 138px;
}
.qq {
  color: #5e5e5e;
}