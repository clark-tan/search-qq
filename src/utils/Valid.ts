
export function isValidQQ(qq: string): boolean {
	return /^[1-9]\d{4,11}$/.test(qq);
}