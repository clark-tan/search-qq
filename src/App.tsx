import React, { useEffect, useState } from "react";
import { isValidQQ }  from './utils/Valid';
import "./App.css";
import SearchResults from "./components/SearchResults";
import { useDebounce } from "./utils/Util";

type QQInfo = {
	name: string;
	qlogo: string;
	qq: string;
};

function App() {
	const [loading, setLoading] = useState(false);
	const [error, setError] = useState<string>("");
	const [data, setData] = useState<QQInfo | null>(null);
	const [search, setSearch] = useState("");

	const onChange = (e: React.ChangeEvent<HTMLInputElement>) => {
		setSearch(e.target.value.trim());
	};

	const handleSearch = (value: string) => {
		if (value.length === 0) {
			setLoading(false);
			setData(null);
			setError("");
			return;
		}

		if (!isValidQQ(value)) {
			setData(null);
			setError("QQ号应该为4位以上的数字！");
			return;
		}

		setError("");
		setLoading(true);
		fetch(`https://api.uomg.com/api/qq.info?qq=${value}`)
			.then((res) => res.json())
			.then((data) => {
				if (data.code !== 1) {
					setError(data.msg);
					setData(null);
				} else {
					setError("");
					setData({
						name: data.name,
						qlogo: data.qlogo,
						qq: data.qq,
					});
				}
			})
			.finally(() => setLoading(false));
	};

	const debounceSearch = useDebounce(search, 500);

	useEffect(() => { handleSearch(debounceSearch); }, [debounceSearch]);

	return (
		<div className="App">
			<div className="title">QQ号查询</div>
			<input
				className="input"
				type="text"
				placeholder="QQ号"
				value={search}
				onChange={onChange}
			/>
			<SearchResults loading={loading}  data={data} error={error}/>
		</div>
	);
}

export default App;
