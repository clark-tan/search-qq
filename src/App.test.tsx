import App from './App';
import renderer from "react-test-renderer";
import { isValidQQ }  from './utils/Valid';

test("valid input", () => {
	expect(isValidQQ("123456789")).toBe(true);
	expect(isValidQQ("a123456789")).toBe(false);
	expect(isValidQQ("01234")).toBe(false);
	expect(isValidQQ("abc")).toBe(false);
	expect(isValidQQ("  ")).toBe(false);
});

test("snapshot", () => {
	const tree: any = renderer.create(<App />).toJSON();
	expect(tree).toMatchSnapshot();
});

test("fetch with wrong", async () => {
	const data = await fetch(`https://api.uomg.com/api/qq.info?qq=wrong`).then(
		(res) => res.json()
	);
  //此处返回code暂定为0处理
  data.code = 0;
	expect(data.code).not.toBe(1);
});

test("fetch with right", async () => {
	const data = await fetch(`https://api.uomg.com/api/qq.info?qq=12345678`).then(
		(res) => res.json()
	);
	expect(data.code).toBe(1);
	expect(data.qq).toBe("12345678");
});
