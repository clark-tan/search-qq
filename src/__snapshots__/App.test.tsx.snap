// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div
  className="App"
>
  <div
    className="title"
  >
    QQ号查询
  </div>
  <input
    className="input"
    onChange={[Function]}
    placeholder="QQ号"
    type="text"
    value=""
  />
  <div
    className="result"
  >
    <div
      className="nodata"
    />
  </div>
</div>
`;
